lib/ansible/modules/storage/glusterfs/glusterfs_volume.py

```diff
@@ -0,0 +1,410 @@
+            option_str = self.brick_ops()
+        if self.action == 'profile':
+            option_str = self._validated_params('profile_state')
+        if self.action == 'set':
+            option_str = self._validated_params('key') + " "
+            option_str += self._validated_params('value')
+        if self.action == 'barrier':
+            option_str = self._validated_params('barrier_state')
+        if self.action == 'bitrot':
+            if self.module.params['scrub']:
+                option_str = 'scrub ' + self._validated_params('scrub')
+            elif self.module.params['scrub_throttle']:
+                option_str = 'scrub-throttle '
+                option_str += self._validated_params('scrub_throttle')
+            elif self.module.params['scrub_frequency']:
+                option_str = 'scrub-frequency '
+                option_str += self._validated_params('scrub_frequency')
+            elif self.module.params['bitrot_daemon']:
+                option_str = self._validated_params('bitrot_daemon')
+
+        rc, output, err = self.call_gluster_cmd('volume', self.action,
+                                               volume, option_str, self.force)
+        self._get_output(rc, output, err)
+
+    def rebalance_volume(self):
+        state = self._validated_params('rebal_state')
+        if state not in ['start', 'stop', 'fix-layout']:
+            self.module.fail_json(rc=20,msg="Invalid state for rebalance action. \n" \
+                    "Available options are: start, stop, fix-layout.")
+        if state != 'start':
+            self.force = ''
+        if state == 'fix-layout':
+            return state + ' start '
+        else:
+            return state
+
+    def create_params_dict(self, param_list):
+        return OrderedDict((param, self.get_playbook_params(param))
+                for param in param_list)
+
+    def call_gluster_cmd(self, *args, **kwargs):
+        params = ' '.join(opt for opt in args)
+        key_value_pair = ' '.join(' %s %s ' % (key, value)
+                for key, value in kwargs)
+        return self._run_command('gluster', ' ' + params + ' ' + key_value_pair)
+
+    def _get_output(self, rc, output, err):
+        changed = 0 if rc else 1
+        if not rc:
+            self.module.exit_json(rc=rc, stdout=output, changed=changed)
+        else:
+            self.module.fail_json(msg=err, rc=rc)
+
+    def _run_command(self, op, opts):
+        cmd = self.module.get_bin_path(op, True) + opts + ' --mode=script'
+        return self.module.run_command(cmd)
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(required=True,
+                       choices=["absent", "present", "remove-brick", "started",
+                                "stopped", "replace-brick", "rebalance",
+                                "add-brick", "profile", "set", "barrier",
+                                "bitrot" ]),
+            hosts=dict(),
```

Comment:
hosts is required as per documentation but does not reflect here.

Reply:
Ack! added required flags for both 'hosts' and 'volname'. Thanks.